aws/resource_aws_appstream_fleet.go

```diff
@@ -0,0 +1,588 @@
+		log.Printf("[WARN] Appstream Fleet (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error reading Appstream Fleet (%s): %w", d.Id(), err))
+	}
+
+	for _, v := range resp.Fleets {
+		d.Set("name", v.Name)
+		d.Set("name_prefix", naming.NamePrefixFromName(aws.StringValue(v.Name)))
+
+		if err = d.Set("compute_capacity", flattenComputeCapacity(v.ComputeCapacityStatus)); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting `%s` for AppStream Fleet (%s): %w", "compute_capacity", d.Id(), err))
+		}
+		if err = d.Set("domain_join_info", flattenDomainInfo(v.DomainJoinInfo)); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting `%s` for AppStream Fleet (%s): %w", "domain_join_info", d.Id(), err))
+		}
+
+		d.Set("description", v.Description)
+		d.Set("display_name", v.DisplayName)
+		d.Set("disconnect_timeout_in_seconds", v.DisconnectTimeoutInSeconds)
+		d.Set("idle_disconnect_timeout_in_seconds", v.IdleDisconnectTimeoutInSeconds)
+		d.Set("enable_default_internet_access", v.EnableDefaultInternetAccess)
+		d.Set("fleet_type", v.FleetType)
+		d.Set("image_name", v.ImageName)
+		d.Set("image_arn", v.ImageArn)
+		d.Set("iam_role_arn", v.IamRoleArn)
+		d.Set("stream_view", v.StreamView)
+		d.Set("arn", v.Arn)
+		d.Set("created_time", aws.TimeValue(v.CreatedTime).Format(time.RFC3339))
+
+		d.Set("instance_type", v.InstanceType)
+		d.Set("max_user_duration_in_seconds", v.MaxUserDurationInSeconds)
+		if err = d.Set("vpc_config", flattenVpcConfig(v.VpcConfig)); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting `%s` for AppStream Fleet (%s): %w", "vpc_config", d.Id(), err))
+		}
+
+		d.Set("state", v.State)
+
+		tg, err := conn.ListTagsForResource(&appstream.ListTagsForResourceInput{
+			ResourceArn: v.Arn,
+		})
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("error listing stack tags for AppStream Stack (%s): %w", d.Id(), err))
+		}
+		if tg.Tags == nil {
+			log.Printf("[DEBUG] Apsstream Stack tags (%s) not found", d.Id())
+			return nil
+		}
+		tags := keyvaluetags.AppstreamKeyValueTags(tg.Tags).IgnoreAws().IgnoreConfig(ignoreTagsConfig)
+
+		if err = d.Set("tags", tags.RemoveDefaultConfig(defaultTagsConfig).Map()); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting `%s` for AppStream Stack (%s): %w", "tags", d.Id(), err))
+		}
+
+		if err = d.Set("tags_all", tags.Map()); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting `%s` for AppStream Stack (%s): %w", "tags_all", d.Id(), err))
+		}
+	}
+	return nil
+}
+
+func resourceAwsAppStreamFleetUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).appstreamconn
+	input := &appstream.UpdateFleetInput{
+		Name: aws.String(d.Id()),
+	}
+
+	// Stop fleet workflow
+	_, err := conn.StopFleetWithContext(ctx, &appstream.StopFleetInput{
```

Comment:
Since many updates can be done without stopping, it would be cool to check if a stop in necessary (probably want to double check these): 

| Attrib | Stop? |
| --- | --- |
| DisplayName, ComputeCapacity, ImageARN, ImageName, IdleDisconnectTimeoutInSeconds,  DisconnectTimeoutInSeconds | No |
| Description, DomainJoinInfo, EnableDefaultInternetAccess, IamRoleArn, InstanceType, MaxUserDurationInSeconds, StreamView, VpcConfig | Yes |
| Name, NamePrefix | ForceNew |